examples/platforms/eagle/radio.c

```diff
@@ -0,0 +1,791 @@
+    bool         rval = false;
+    otMacAddress src;
+
+    otEXPECT(otMacFrameIsDataRequest(aFrame));
+    otEXPECT_ACTION(sSrcMatchEnabled, rval = true);
+    otEXPECT(otMacFrameGetSrcAddr(aFrame, &src) == OT_ERROR_NONE);
+
+    switch (src.mType)
+    {
+    case OT_MAC_ADDRESS_TYPE_SHORT:
+        rval = utilsSoftSrcMatchShortFindEntry(src.mAddress.mShortAddress) >= 0;
+        break;
+    case OT_MAC_ADDRESS_TYPE_EXTENDED:
+    {
+        otExtAddress extAddr;
+
+        ReverseExtAddress(&extAddr, &src.mAddress.mExtAddress);
+        rval = utilsSoftSrcMatchExtFindEntry(&extAddr) >= 0;
+        break;
+    }
+    default:
+        break;
+    }
+
+exit:
+    return rval;
+}
+
+uint8_t rf_rssi_to_lqi(int8_t aRss)
+{
+    int8_t  aNoiseFloor = -99;
+    int8_t  linkMargin  = aRss - aNoiseFloor;
+    uint8_t aLinkMargin;
+    uint8_t threshold1, threshold2, threshold3;
+    uint8_t linkQuality = 0;
+
+    if (linkMargin < 0 || aRss == OT_RADIO_RSSI_INVALID)
+    {
+        linkMargin = 0;
+    }
+
+    aLinkMargin = (uint8_t)(linkMargin);
+
+    threshold1 = 2;
+    threshold2 = 10;
+    threshold3 = 20;
+
+    if (aLinkMargin > threshold3)
+    {
+        linkQuality = 3;
+    }
+    else if (aLinkMargin > threshold2)
+    {
+        linkQuality = 2;
+    }
+    else if (aLinkMargin > threshold1)
+    {
+        linkQuality = 1;
+    }
+
+    return linkQuality;
+}
+
+void eagleRadioInit(void)
+{
+    int           i;
+    otRadioFrame *p;
+
+    sTransmitFrame.mLength = 0;
+    sTransmitFrame.mPsdu   = sTransmitPsdu;
+
+    for (i = 0; i < RX_FRAME_SLOT_NUM; i++)
+    {
+        p          = (otRadioFrame *)rx_frame_slots[i];
+        p->mLength = 0;
+        p->mPsdu   = &(rx_frame_slots[i][MEM_ALIGN_SIZE(sizeof(otRadioFrame))]);
+    }
+    p          = (otRadioFrame *)rx_frame;
+    p->mLength = 0;
+    p->mPsdu   = &(rx_frame[MEM_ALIGN_SIZE(sizeof(otRadioFrame))]);
+
+    sAckFrame.mLength = 0;
+    sAckFrame.mPsdu   = sAckPsdu;
+
+    rf_mode_init();
+    rf_set_zigbee_250K_mode();
+    rf_set_power_level(RF_POWER_P9p11dBm);
+    rf_set_tx_dma(2, 256);
+    rf_set_rx_dma(rx_buffer, 3, 256);
+    plic_interrupt_enable(IRQ15_ZB_RT);
+    rf_set_irq_mask(FLD_RF_IRQ_RX | FLD_RF_IRQ_TX);
+}
+
+void EagleRxTxIntHandler()
+{
+    if (rf_get_irq_status(FLD_RF_IRQ_RX))
+    {
+        dma_chn_dis(DMA1);
+        rf_clr_irq_status(FLD_RF_IRQ_RX);
+
+        if (rf_zigbee_packet_crc_ok(rx_buffer) /*&& rf_zigbee_packet_length_ok(rx_buffer)*/)
+        {
+            uint8_t       length;
+            otRadioFrame *rx_frame_ptr;
+            int           i;
+
+            // radioProcessFrame();
+            otEXPECT(sState == OT_RADIO_STATE_RECEIVE || sState == OT_RADIO_STATE_TRANSMIT);
+            length = rx_buffer[4];
+            otEXPECT(IEEE802154_MIN_LENGTH <= length && length <= IEEE802154_MAX_LENGTH);
+
+            if (empty > 0)
+            {
+            }
+            else
+            {
+                goto exit;
+            }
```

Comment:
OpenThread [code style](https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md#language-independent) suggest us do not use the local `goto`.  You may follow the following comments to change the code. 

Add the following code between line 637 and 638:
```
bool dmaChnEn = false;
```
Lines between 654 and 660 are changed to:
```
otEXPECT_ACTION(empty > 0, dmaChnEn=true );
```

Lines between 677 and 681 are changed to:
```
otEXPECT_ACTION(otMacFrameDoesAddrMatch(rx_frame_ptr, sPanid, sShortAddress, &sExtAddress), dmaChnEn = true );
```

Lines between 717 and 781 are changed to:
```
dmaChnEn=true;
```
Add the following code to line 732:
```
exit:
      if (dmaChnEn)
      {
            dma_chn_en(DMA1);
      }

```